src/ripple/protocol/SecretKey.h

```diff
@@ -35,10 +35,13 @@ namespace ripple {
 class SecretKey
```

Comment:
No change needed, but here's a random thought: I notice that in another part of this commit we derive from `std::array`. _If_ that is OK (@HowardHinnant?) then we may be able to significantly simply `SecretKey`. Given:

    class SecretKey : public std::array<std::uint8_t, 32>

then there's no need to have `data()`, `size()`, `begin()`, `end()`, `cbegin()` and `cend()`. In fact, we could almost get away with:

    using SecretKey = std::array<std::uint8_t, 32>;

Reply:
Normally I wouldn't get too worked up about this.  But with `SecretKey` I think it is worth the effort to keep it its own type.  It would probably never happen, but I would hate for there to be an even remote chance of accidentally sending a `SecretKey` to a function that was expecting a `std::array`.

Reply:
Agreed. I was mostly curious about the semantics of deriving from `std::array`. Is that legal?

Reply:
It's legal.  But not generally recommended.  I consider it ok if the derived class _is_ an `array`, and _never_ expects to be destructed via a pointer to an `array`.